File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int goodNodes(TreeNode* root) {
        int max = -1e4, count = 0;
        checkGooeNode(root, max, count);
        
        return count;                        
    }
private:
    void checkGooeNode(TreeNode* root, int max, int& count) {
        if (root->val >= max) {
            max = root->val;
            count++;
            cout<<root->val<<endl;
        }
        
        if (root->left) checkGooeNode(root->left, max, count);
        if (root->right) checkGooeNode(root->right, max, count);
        
    }
};

File: 1448-count-good-nodes-in-binary-tree/1448-count-good-nodes-in-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        self.count = 0
        maximum = -1e4
        self.checkGoodNode(root, maximum)
        
        return self.count
                
    
    def checkGoodNode(self, root, maximum):
        if root:
            if root.val >= maximum:
                self.count += 1
                maximum = root.val
            self.checkGoodNode(root.left, maximum)
            self.checkGoodNode(root.right, maximum)
        